### PR TITLE
types: migrate test-infra to testify for errors_test.go

### LINKED ISSUE #28001: migrate test-infra to testify for `errors_test.go`

### PR DESCRIPTION
**Pull request by hanlins**
Signed-off-by: Hanlin Shi <shihanlin9@gmail.com>

### What problem does this PR solve?

Issue Number: close #28001 

```release-note
NONE
```

### CODE REVIEW

File: types/errors_test.go [hunk 1]
```diff
@@ -50,8 +48,11 @@ func (s testErrorSuite) TestError(c *C) {
 		ErrInvalidWeekModeFormat,
 		ErrWrongValue,
 	}
-	for _, err := range kvErrs {
-		code := terror.ToSQLError(err).Code
-		c.Assert(code != mysql.ErrUnknown && code == uint16(err.Code()), IsTrue, Commentf("err: %v", err))
-	}
+
+	t.Run("ValidateErrorCodes", func(t *testing.T) {
```
Comment (xhebox) [inline types/errors_test.go:52]: You probably should do:

```
for _, err := range kvErrs {
     t.Run("ValidateErrorCodes", func(t *testing.T) {
			code := terror.ToSQLError(err).Code
			require.Equal(t, uint16(err.Code()), code, fmt.Sprintf("err: %v", err))
  })
}
```

But my recommendation would be:

```
func TestError(t *testing.T) {
t.Parallel()
```

There is no need to parallel tests for such a small test

Comment (hanlins) [reply types/errors_test.go]: Done, PTAL

File: types/errors_test.go [hunk 2]
```diff
@@ -50,8 +48,12 @@ func (s testErrorSuite) TestError(c *C) {
 		ErrInvalidWeekModeFormat,
 		ErrWrongValue,
 	}
+
+	t.Parallel()
 	for _, err := range kvErrs {
-		code := terror.ToSQLError(err).Code
-		c.Assert(code != mysql.ErrUnknown && code == uint16(err.Code()), IsTrue, Commentf("err: %v", err))
+		t.Run("ValidateErrorCodes", func(t *testing.T) {
```
Comment (xhebox) [inline types/errors_test.go:54]: You don't need `t.Run("ValidateErrorCodes", func(t *testing.T) {` if there is `t.Parallel()` outside. You could refer to the document of go for more description, how sub tests and parallel tests work.

Comment (xhebox) [reply types/errors_test.go:54]: We create sub tests to parallel tests.

**Review by tisonkun — APPROVED** [review]
I just push a follow up for you @hanlins . It won't be that complex.

**Review by xhebox — APPROVED** [review]

**Pull request merged**